ddtrace/tracer/option.go

```diff
@@ -96,7 +97,7 @@ func defaults(c *config) {
 	if v := os.Getenv("DD_DOGSTATSD_PORT"); v != "" {
 		statsdPort = v
 	}
-	c.dogstatsdAddr = net.JoinHostPort(statsdHost, statsdPort)
+	WithDogstatsdAddress(net.JoinHostPort(statsdHost, statsdPort))
```

Comment:
It's funny that you're proposing this, because we've actually had this discussion internally and we've decided to keep it as a direct assignment rather than go through the option. It's more obvious and clear what's going on in most cases (made exception of the global tags further down).

```
c.dogstatsdAddr = net.JoinHostPort(statsdHost, statsdPort)
```
Looks better than the more "magical":
```
WithDogstatsdAddress(net.JoinHostPort(statsdHost, statsdPort))(c)
```
What do you think?

Reply:
@gbbr  I think consistency should win here, having 2 ways of applying a setting seems dangerous to me :)